store/postgres/src/store.rs

```diff
@@ -823,6 +828,66 @@ impl SubgraphDeploymentStore for Store {
                 .map_err(|()| format_err!("deployment event notification listener failed")),
         )
     }
+
+    fn is_deployed(&self, id: &SubgraphId) -> Result<bool, Error> {
+        use db_schema::subgraph_deployments::dsl::*;
+
+        // The subgraph of subgraphs is always deployed.
+        if id == &SubgraphId::new(SUBGRAPHS_ID).unwrap() {
+            return Ok(true);
+        }
+
+        let deployment_count = subgraph_deployments
+            .filter(subgraph_id.eq(id.to_string()))
+            .execute(&*self.conn.lock().unwrap())?;
+        Ok(match deployment_count {
+            0 => false,
+            1 => true,
+            _ => unreachable!(), // The `subgraph_id` column is `UNIQUE`
+        })
+    }
+
+    fn subgraph_schema(&self, subgraph_id: SubgraphId) -> Result<Schema, Error> {
+        if let Some(schema) = self.schema_cache.lock().unwrap().get(&subgraph_id) {
+            trace!(self.logger, "schema cache hit"; "id" => subgraph_id.to_string());
+            return Ok(schema.clone());
+        }
+
+        let raw_schema = if subgraph_id == SubgraphId::new(SUBGRAPHS_ID).unwrap() {
+            // The subgraph of subgraphs schema is built-in.
+            include_str!("subgraphs.graphql").to_owned()
+        } else {
+            let manifest_entity = self
+                .get(EntityKey {
+                    subgraph_id: SubgraphId::new(SUBGRAPHS_ID).unwrap(),
+                    entity_type: MANIFEST_ENTITY_TYPENAME.to_owned(),
+                    entity_id: format!("{}{}", subgraph_id, MANIFEST_SUFFIX),
```

Comment:
How about extracting these to functions in schema.rs that either create the entity ID, or create the whole EntityKey? `SubgraphManifestEntity::id(subgraph_id)`, `SubgraphManifestEntity::entiyt_key(subgraph_id)` or something